pkg/services/sqlstore/migrations/accesscontrol/datasource_drilldown_removal.go

```diff
@@ -0,0 +1,42 @@
+package accesscontrol
+
+import (
+	"fmt"
+
+	"xorm.io/xorm"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+const (
+	datasourceDrilldownRemoval = "delete all drilldown actions, their assignments, and related roles"
+)
+
+func AddDatasourceDrilldownRemovalMigration(mg *migrator.Migrator) {
+	mg.AddMigration(datasourceDrilldownRemoval, &datasourceDrilldownRemovalMigrator{})
+}
+
+type datasourceDrilldownRemovalMigrator struct {
+	migrator.MigrationBase
+}
+
+func (m *datasourceDrilldownRemovalMigrator) SQL(dialect migrator.Dialect) string {
+	return CodeMigrationSQL
+}
+
+func (m *datasourceDrilldownRemovalMigrator) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
+	result, err := sess.Exec("DELETE FROM permission WHERE action LIKE ?", "%datasources:drilldown%")
+	if err != nil {
+		mg.Logger.Error("Failed to delete datasources:drilldown permissions", "error", err)
+		// This is a critical step, but we'll continue with the migration
```

Comment:
What's the thinking behind not erroring out here?

Reply:
this was because of the role assignments. and thinking that we would encounter problems for some of the instances where we do not have the permission yet